VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java

```diff
@@ -54,27 +66,48 @@ public void call(Session session, SessionState state, Exception exception) {
         }
     };
 
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    ProfileModel profile;
+
+    ImageView imageView;
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         uiLifecycleHelper = IUiLifecycleHelper.Factory.getInstance(getActivity(), callback);
         uiLifecycleHelper.onCreate(savedInstanceState);
-    }
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-            Bundle savedInstanceState) {
-        super.onCreateView(inflater, container, savedInstanceState);
         Context context = getActivity().getBaseContext();
-
         socialPref = new PrefManager(context, PrefManager.Pref.FEATURES);
-
         pref = new PrefManager(context, PrefManager.Pref.LOGIN);
+        profile = pref.getCurrentUserProfile();
+        if (config.isUserProfilesEnabled() && profile != null && profile.username != null) {
+            getAccountTask = new GetAccountTask(getActivity(), profile.username) {
+                @Override
+                protected void onSuccess(Account account) throws Exception {
+                    final ProfileImage profileImage = account.getProfileImage();
+                    Glide.with(NavigationFragment.this)
+                            .load(profileImage.getImageUrlLarge())
+                                    .skipMemoryCache(true)
+                            .into(imageView);
+                }
+            };
+            getAccountTask.execute();
+        }
+    }
 
-        View layout = inflater.inflate(R.layout.drawer_navigation, null);
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View layout = inflater.inflate(R.layout.drawer_navigation, container, false);
 
         TextView name_tv = (TextView) layout.findViewById(R.id.name_tv);
         TextView email_tv = (TextView) layout.findViewById(R.id.email_tv);
+        imageView = (ImageView) layout.findViewById(R.id.profile_image);
```

Comment:
I suggest that we use the Loader framework. It has been designed to
integrate with the UI lifecycle, and handle these issues automatically.
Here's an implementation that could be used with Retrofit 2.0:
https://gist.github.com/1zaman/dab740c55810de693054

On Fri, Sep 25, 2015 at 8:39 PM, Brian Guertin <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java
> <https://github.com/edx/edx-app-android/pull/372#discussion_r40443398>:
>
> >
> >          TextView name_tv = (TextView) layout.findViewById(R.id.name_tv);
> >          TextView email_tv = (TextView) layout.findViewById(R.id.email_tv);
> > +        imageView = (ImageView) layout.findViewById(R.id.profile_image);
>
> Cancelling the task does not guarantee that already sent results would not
> be delivered
>
> That's unfortunate. Not to mention SafeAsyncTask is deprecated. We should
> move on to something better soon.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/372/files#r40443398>.
>